src/bentoml/_internal/frameworks/diffusers.py

```diff
@@ -0,0 +1,355 @@
+            ``str`` or ``int``.
+
+    Returns:
+        A :obj:`~bentoml.Model` instance referencing a saved model in the local BentoML
+        model store.
+
+    Example:
+
+    .. code-block:: python
+
+        import bentoml
+
+        bentoml.diffusers.import_model(
+            'my_sd15_model',
+            "runwayml/stable-diffusion-v1-5",
+            signatures={
+                "__call__": {"batchable": False},
+            }
+        )
+    """
+    context = ModelContext(
+        framework_name="diffusers",
+        framework_versions={"diffusers": diffusers.__version__},
+    )
+
+    if signatures is None:
+        signatures = {
+            "__call__": {"batchable": False},
+        }
+        logger.info(
+            'Using the default model signature for diffusers (%s) for model "%s".',
+            signatures,
+            name,
+        )
+
+    with bentoml.models.create(
+        name,
+        module=MODULE_NAME,
+        api_version=API_VERSION,
+        signatures=signatures,
+        labels=labels,
+        options=None,
+        custom_objects=custom_objects,
+        external_modules=external_modules,
+        metadata=metadata,
+        context=context,
+    ) as bento_model:
+
+        diffusion_model_dir = bento_model.path_of(DIFFUSION_MODEL_FOLDER)
+        ignore = shutil.ignore_patterns(".git")
+
+        if os.path.isdir(model_name_or_path):
+            src_dir = model_name_or_path
+
+        else:
+
+            try:
+                from huggingface_hub import snapshot_download
+            except ImportError:  # pragma: no cover
+                raise MissingDependencyException(
+                    "'huggingface_hub' is required in order to download pretrained diffusion models, install with 'pip install huggingface-hub'. For more information, refer to https://huggingface.co/docs/huggingface_hub/quick-start",
+                )
+
+            src_dir = snapshot_download(
+                model_name_or_path,
+                proxies=proxies,
+                revision=revision,
+            )
+
+        model_config_file = os.path.join(src_dir, DIFFUSION_MODEL_CONFIG_FILE)
+        if not os.path.exists(model_config_file):
+            raise BentoMLException(f'artifact "{src_dir}" is not a Diffusion model')
+
+        shutil.copytree(src_dir, diffusion_model_dir, symlinks=False, ignore=ignore)
+
+        return bento_model
+
+
+def get_runnable(bento_model: bentoml.Model) -> t.Type[bentoml.Runnable]:
+    """
+    Private API: use :obj:`~bentoml.Model.to_runnable` instead.
+    """
+
+    partial_kwargs: t.Dict[str, t.Any] = bento_model.info.options.partial_kwargs  # type: ignore
+    pipeline_class: type[diffusers.DiffusionPipeline] = (
+        bento_model.info.options.pipeline_class or diffusers.StableDiffusionPipeline
+    )
+    scheduler_class: type[
+        diffusers.SchedulerMixin
+    ] | None = bento_model.info.options.scheduler_class
```

Comment:
This two will fail when dumping these classes to yaml. I think you might have to write custom handler for `bentoml_cattr`.

Reply:
Currently saving won't save these options. So we may need to worry about pickling.